quote-book/synchronized/src/lib.rs

```diff
@@ -2,24 +2,80 @@
 
 use deqs_quote_book_api::{Error, Pair, Quote, QuoteBook, QuoteId};
 use mc_blockchain_types::BlockIndex;
+use mc_common::logger::{log, Logger};
 use mc_crypto_ring_signature::KeyImage;
-use mc_ledger_db::Ledger;
+use mc_ledger_db::{Error as LedgerError, Ledger};
 use mc_transaction_extra::SignedContingentInput;
-use std::ops::RangeBounds;
+use std::{
+    ops::RangeBounds,
+    sync::{
+        atomic::{AtomicBool, AtomicU64, Ordering},
+        Arc,
+    },
+    thread::{Builder as ThreadBuilder, JoinHandle},
+    time::Duration,
+};
 
 /// A wrapper for a quote book implementation that syncs quotes with the ledger
 #[derive(Clone)]
-pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + 'static> {
+pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> {
     /// Quotebook being synchronized to the ledger by the Synchronized Quotebook
     quote_book: Q,
 
+    /// Ledger
     ledger: L,
+
+    /// Index of the highest processed block index by the background thread.
+    highest_processed_block_index: Arc<AtomicU64>,
+
+    /// Thread management. This is being held purely for the destructor so that
+    /// the SyncThread will be dropped after all the Quotebooks using it.
+    _thread_manager: Arc<SyncThread>,
 }
 
 impl<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> SynchronizedQuoteBook<Q, L> {
     /// Create a new Synchronized Quotebook
-    pub fn new(quote_book: Q, ledger: L) -> Self {
-        Self { quote_book, ledger }
+    pub fn new(
+        quote_book: Q,
+        ledger: L,
+        remove_quote_callback: RemoveQuoteCallback,
+        logger: Logger,
+    ) -> Self {
+        let ledger_clone = ledger.clone();
+        let quote_book_clone = quote_book.clone();
+        let highest_processed_block_index = Arc::new(AtomicU64::new(0));
+        let thread_highest_processed_block_index = highest_processed_block_index.clone();
+        let stop_requested = Arc::new(AtomicBool::new(false));
+        let thread_stop_requested = stop_requested.clone();
```

Comment:
it would be nice to move some more of these guts inside a `SyncThread::new` or `SyncThread::start` function later, but it's mostly cosmetic